go/storage/mkvs/syncer/proof.go

```diff
@@ -33,21 +44,37 @@ type proofNode struct {
 
 // ProofBuilder is a Merkle proof builder.
 type ProofBuilder struct {
-	root     hash.Hash
-	subtree  hash.Hash
-	included map[hash.Hash]*proofNode
-	size     uint64
+	proofVersion uint16
+	root         hash.Hash
+	subtree      hash.Hash
+	included     map[hash.Hash]*proofNode
+	size         uint64
 }
 
 // NewProofBuilder creates a new Merkle proof builder for the given root.
 func NewProofBuilder(root, subtree hash.Hash) *ProofBuilder {
 	return &ProofBuilder{
-		root:     root,
-		subtree:  subtree,
-		included: make(map[hash.Hash]*proofNode),
+		proofVersion: LatestProofVersion,
+		root:         root,
+		subtree:      subtree,
+		included:     make(map[hash.Hash]*proofNode),
 	}
 }
 
+// NewProofBuilderForVersion creates a new Merkle proof builder for the given root
+// in a given proof version format.
+func NewProofBuilderForVersion(root, subtree hash.Hash, proofVersion uint16) (*ProofBuilder, error) {
+	if proofVersion < MinimumProofVersion || proofVersion > LatestProofVersion {
+		return nil, fmt.Errorf("proof builder: invalid proof version: %d", proofVersion)
```

Comment:
```suggestion
		return nil, fmt.Errorf("proof: invalid proof version: %d", proofVersion)
```
I prefer not to return errors in constructors, if possible. As the programmer will select the version, we can maybe add a constructor for each version, e.g. `NewProofBuilder`, `NewProofBuilderV0`, `NewProofBuilderV1`, ... and use this constructor only as a helper (if needed).

Reply:
> ..As the programmer will select the version. 

That's not true in this case. The ReadSyncer request will contain the proof version (for backwards compatibility so that existing runtimes will get the current versions of proofs returned).

In places where the programmer selects the version the `NewProofBuilder` constructor is used (which also defaults to latest version).